server/products.js

```diff
@@ -17,17 +22,71 @@ module.exports = require('express').Router()
   //  only admins should be able to create products
   .post('/', (req, res, next) => {
     Product.create(req.body)
-    .then(product => res.json(product)
-    .catch(next))
+    .then(product => res.status(201).json(product))
+    .catch(next)
   })
   //  only admins should be able to edit products
+  // .put('/:id', (req, res, next) => {
+  //   Product.update(req.body, {
+  //     where: {
+  //       id: req.params.id
+  //     },
+  //     returning: true,
+  //     plain: true
+  //   })
+  // .then(([count, product]) => res.status(201).json(product))
+  // .catch(next)
+  // })
+
   .put('/:id', (req, res, next) => {
-    Product.update(req.body, {where: {id: req.params.id}})
-    .then(product => res.json(product[1]))
-  .catch(next)
+    Promise.all([
+      Product.update(req.body, {
+        where: {
+          id: req.params.id
+        },
+        returning: true,
+        plain: true
+      }),
+      Order.findAll({
+        where: {
+          status: 'cart'
+        },
+      })
+    ])
+    .then(([[count, product], orders]) => {
+      console.log('product', product)
+      orders.forEach(order =>
+        order.setProducts(product, {price: product.price}))
+    })
+    .then(() => res.sendStatus(201))
+    .catch(next)
   })
+
   //  only admins should be able to delete products
   .delete('/:id', (req, res, next) => {
     Product.destroy({where: {id: req.params.id}})
     .then(() => res.sendStatus(202))
   })
+
```

Comment:
these routes are probably okay, but the uniform nature of them sorta leads me to believe that these haven't been changed. I suspect that more work will need to be done here to make it working.